scripts/installer.sh

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env bash
+#
+# Copyright contributors to the Hyperledger Fabric Operator project
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at:
+#
+# 	  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# variants for options
+UPGRADE_U4A=${UPGRADE_U4A:-"NO"}
+INSTALL_U4A=${INSTALL_U4A:-"NO"}
+
+REMOVE_FABRIC_OPERATOR=${REMOVE_FABRIC_OPERATOR:-"NO"}
+UPGRADE_FABRIC_OPERATOR=${UPGRADE_FABRIC_OPERATOR:-"NO"}
+IGNORE_FABRIC_OPERATOR=${IGNORE_FABRIC_OPERATOR:-"YES"}
+
+TIMEOUT=${TIMEOUT:-"1200s"}
+
+echo "checking option params..."
+
+# Get install options
+for i in "$@"; do
+	echo $i
+	if [ $i == "--all" ];
+	then
+		echo "will install all parts by default"
+		INSTALL_U4A="YES"
+		IGNORE_FABRIC_OPERATOR="NO"
+	elif [ $i == "--u4a" ];
+	then 
+		echo "will install u4a components"
+		INSTALL_U4A="YES"
+	elif [ $i == "--baas" ];
+	then
+		echo "will install fabric-operator"
+		IGNORE_FABRIC_OPERATOR="NO"
+	elif [ $i == "--addon" ];
+	then
+		echo "will install addons"
+		# TODO: future components integration
+	elif [ $i == "--up-all" ];
+	then
+		echo "will upgrade all components"
+		UPGRADE_U4A="YES"
+		UPGRADE_FABRIC_OPERATOR="YES"
+		# TODO: upgrade other components
+	elif [ $i == "--up-baas" ];
+	then
+		echo "will upgrade fabric-operator"
+		UPGRADE_FABRIC_OPERATOR="YES"
+	elif [ $i == "--rm-baas" ];
+	then
+		echo "will remove fabric-operator"
+		REMOVE_FABRIC_OPERATOR="YES"
+	else
+		echo "param error, no changes applied"
+	fi
+done
+
+function debug() {
+	kubectl describe po -A
+	kubectl get po -A
+	exit 1
+}
+trap debug ERR
+
+# step 1. create namespace
+if [ $INSTALL_U4A == "YES" ]; then
+	kubectl create namespace u4a-system
+fi
+
+# step 2. get node name and node ip
+ingressNode="kind-worker"
+kubeProxyNode="kind-worker2"
+ingressNodeIP=$(kubectl get node ${ingressNode} -owide | grep -v "NAME" | awk '{print $6}')
+kubeProxyNodeIP=$(kubectl get node ${kubeProxyNode} -owide | grep -v "NAME" | awk '{print $6}')
+
+echo "node info:"
+kubectl get node -o wide
+echo "ingressNodeIp is: ${ingressNodeIP}"
+echo "kubeProxyNodeIP is: ${kubeProxyNodeIP}"
+export ingressNodeIP=${ingressNodeIP}
+
+# step 3. repalce ingress node name
+cat ../u4a-component/charts/cluster-component/values.yaml | sed "s/<replaced-ingress-node-name>/${ingressNode}/g" \
```

Comment:
I think it should be changed to absolute path here.

```bash
ABSPATH=$(cd "$(dirname "$0")";pwd)
cat $ABSPATH/u4a-component/xxxx
```

Reply:
The origin `e2e.sh` uses absolute path to co-operate with the `makefile` by `$ make e2e`.  
Maybe adding a similar `make installer [PARAM]` in the `makefile` will solve it ?